packages/components/src/tab-panel/index.tsx

```diff
@@ -104,7 +104,7 @@ export function TabPanel( {
 					( tabs.length > 0 ? tabs[ 0 ].name : undefined )
 			);
 		}
-	}, [ tabs ] );
+	}, [ tabs, selected, initialTabName ] );
```

Comment:
This `find()` on L100 is already done outside of the useEffect on L96:

```js
const newSelectedTab = find( tabs, { name: selected } );
```

Could we perhaps reuse that `selectedTab` variable instead of rerunning the `find()`?

```suggestion
	}, [ tabs, selectedTab, initialTabName ] );
```

Like you say, it's not an expensive operation, but more hygienic and readable nonetheless 😄

Reply:
Good point - though I think we might want to memoize `selectedTab` before introducing it as a dependency on the `useEffect`.

The current dep `selected` is a string, so it won't cause the effect to fire if the value doesn't change. `selectedTab` is an object, so we'd end up firing the effect on each re-render. Updated in 465d9e2 what do you think?

Reply:
Ah ok. Maybe we could pass `selectedTab?.name` instead of `selectedTab` to the useEffect's dep array then, so we don't need to memoize. Would that work?

Reply:
Yes, I think it will! Updated and rebased!